cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/explore/04/traces_details.spec.js

```diff
@@ -0,0 +1,633 @@
+            cy.contains('Filter errors').click();
+
+            // Verify filter badge appears
+            cy.get('[data-test-subj^="filter-badge-"]').should('be.visible');
+
+            // Verify clear filters button appears
+            cy.get('[data-test-subj="clear-all-filters-button"]').should('be.visible');
+
+            // Clear filters
+            cy.get('[data-test-subj="clear-all-filters-button"]').click();
+
+            // Verify filter badge disappears
+            cy.get('[data-test-subj^="filter-badge-"]').should('not.exist');
+          }
+        });
+      });
+
+      it('should handle service legend modal', () => {
+        // Check if service legend button exists
+        cy.get('body').then(($body) => {
+          if ($body.find('button').filter(':contains("Service legend")').length > 0) {
+            // Click service legend button
+            cy.contains('Service legend').click();
+
+            // Verify modal opens
+            cy.get('.euiModal').should('be.visible');
+            cy.contains('Service legend').should('be.visible');
+
+            // Verify service list is shown
+            cy.get('.euiModal .euiText').should('have.length.greaterThan', 0);
+
+            // Close modal
+            cy.get('.euiModal button').contains('Close').click();
+
+            // Verify modal closes
+            cy.get('.euiModal').should('not.exist');
+          }
+        });
+      });
+    });
+
+    describe('Performance and Reliability', () => {
+      it('should handle page resize without breaking gantt chart', () => {
+        // Wait for the main gantt chart panel
+        cy.get('[data-test-subj="span-gantt-chart-panel"]', { timeout: 15000 }).should(
+          'be.visible'
+        );
+
+        // Resize viewport
+        cy.viewport(1200, 800);
+
+        // Wait a moment for resize to take effect
+        cy.wait(1000);
+
+        // Verify gantt chart is still functional
+        cy.get('body').then(($body) => {
+          let svgSelector = '.exploreGanttChart__container svg';
+          if ($body.find('.exploreGanttChart__container svg').length === 0) {
+            svgSelector = '[data-test-subj="span-gantt-chart-panel"] svg';
+          }
+
+          cy.get(svgSelector).should('be.visible');
+          cy.get(`${svgSelector} .mark-rect path`).should('have.length.greaterThan', 0);
+
+          // Verify interactions still work
+          cy.get(`${svgSelector} .mark-rect path`).first().trigger('mousedown', { force: true });
+          cy.url().should('include', 'spanId');
+        });
+
+        // Reset viewport
+        cy.viewport(1280, 720);
+      });
+
+      it('should load trace data within reasonable time', () => {
+        // Verify page loads quickly
+        cy.get('[data-test-subj="span-gantt-chart-panel"]', { timeout: 15000 }).should(
+          'be.visible'
+        );
+
+        // Verify gantt chart and span data is loaded
+        cy.get('body').then(($body) => {
```

Comment:
if i remember correctly. the problem with accessing body is that once you access the body it will access it successfully right away. however if the body was accessed before the elements were ever loaded then it will never refetch the body.

we should avoid access the body like this. it will make the test flakey